### DISCUSSION TITLE
Adding basic Shadows and Outlines to Sprites to mimic common Photoshop funcitonality (extras to be added over time).

### DISCUSSION THREAD

**Discussion by Corso-1** [godotengine/godot-proposals#9441] [2D]
Hiya,
As per the image.

![Photoshop Options for Sprites](https://github.com/godotengine/godot-proposals/assets/161096738/d39fd563-8af7-49d0-a1e7-194b3488efed)

Basically, drop shadows with the ability to skew and offset, simplistic outline and adding Photoshop layer functionality that could be natural for PS users.

Taking the fuss out of adding mini 'shadows' and 'selection' functions. 
Shadows, for 2d to 2.5D effect
Outline, for highlight or select-able user actions.

I do realise everyone is going to scream SHADERS NOW at me, but, there are 2 important reasons for proposing a possible Sprite2dPS. One, because many don't come from a C background, so GDScript is fantastic (easy), but shader language can makle one disillusioned with the lack of Visual Shader tutorials/documentation/implementation, and that code versions often have no //explanation text, and now since upgrades, deciphering some are already broken. As a newbie, I can't pretend to fix them. 
So I suspect, like others, we are hitting a 'Shader' impasse, repeatedly with no real way to fight a way through it. Which would relate to why people regard Shaders as Dark Magic and Hard.

So, the above, the Sprite2dPS would be a standard Sprite2d but with Photoshop-style shader functions as deemed important by users. Layer effects/gradients/etc which could be added over time. 

This would make Photoshop users excited, as with PS, you can make a lot of tricky 'illusions' to fancify (juice up) your work. And being able to find Sprites supporting them would be like bees to honey.

Currently, I cant find working Shader code to mimic shadows, that follow a Global Position Offset, they are all Local, which twist incorrectly with the sprite. So, the workaround was to make a separate shadow sprite and position it underneath the Sprite, with rotation and global offset. Things like this would be ideal for an enhanced Sprite. Same deal with an outline, it's made using a separate image.

Thanks for reading.

**Comment by Calinou** [comment]
- Related to https://github.com/godotengine/godot-proposals/issues/2835.

For performance reasons, we need to keep the default CanvasItem shader as simple as possible. Adding optional features to it would make 2D rendering slower even if the features aren't used in a project.

Instead, we should improve discoverability of user-made shaders as mentioned in https://github.com/godotengine/godot-proposals/issues/96#issuecomment-536303775.

**Reply by Corso-1** [reply]
That would be awesome. Having basic Photoshoppy Layer effects as shaders (we can pick and choose from) would be fantastic. As a beginner, tinkering with something that already exists is an efficeint way to learn too.
